### DISCUSSION TITLE
How to increase/control the height of the output container for revealjs presentations?

### DISCUSSION THREAD

**Discussion by cwsmonden** [quarto-dev/quarto-cli#8528] [Q&A]
### Description

I use Quarto to make slides, with revealjs, for my lectures. As in the example below, sometimes the output is longer than the maximum height of the container that the output appears in. 
I can't find a way to control the height of the container when the format is revealjs. Note there is no problem if the format is HTML. When rendering to HTML I can set the (max) height of the container using **#| attr-output:**  but this doesn't work with revealjs. 
What can I do to control/increase the height of the container for text/console output when using revealjs?

````qmd
---
format: revealjs
---

// Output container not controlled through attr-output 
// the result is a container with a scroll bar

```{r}
#| attr-output: 'style="height: 600px"'
m1 <- lm(data=mtcars, mpg ~ cyl + hp + wt)
summary(m1)
```
````

directly to HTML I can set the container height:

````qmd
---
format: html
---

// Output container can be controlled 
// Full output in the rendered document

```{r}
#| attr-output: 'style="height: 100px"'
m1 <- lm(data=mtcars, mpg ~ cyl + hp + wt)
summary(m1)
```

**Comment by cderv** [comment]
Revealjs framework is quite specific compared to regular HTML. It will automatically resize slide for example. More on that on https://revealjs.com/presentation-size/ and it will also use specific DOM for its slide layout. 

You should try to target the element you want to size using CSS directly. `attr-output` inline style may not apply to the right element for the sizing. 

Did you try already to inspect the HTML and see which is the element you need to style ?

**Comment by remlapmot** [comment]
I have found that to increase the max height of the code output cell in revealjs that you need the following CSS. I believe the current default is 400px, so pick a figure bigger than that.

```css
.cell .cell-output-stdout pre code {
  max-height: 650px; // Adjust this value as needed
  overflow-y: auto; // Add this to handle overflow
}
```

And the max height of the code block of the code (not its output) can be increased by the `code-block-height` argument <https://quarto.org/docs/presentations/revealjs/index.html#code-block-height>.
